cinema/views.py

```diff
@@ -45,9 +76,31 @@ def get_serializer_class(self):
 
 
 class MovieSessionViewSet(viewsets.ModelViewSet):
-    queryset = MovieSession.objects.all()
+    queryset = (
+        MovieSession.objects
+        .select_related("movie", "cinema_hall")
+        .annotate(
+            tickets_available=(
+                F("cinema_hall__rows")
+                * F("cinema_hall__seats_in_row")
+                - Count("tickets") - 1
```

Comment:
The calculation for `tickets_available` subtracts 1, which might not be necessary. Ensure that this subtraction is intentional and correct, as it could lead to incorrect ticket availability counts.